Source/AwsCommonRuntimeKit/crt/ByteBuffer.swift

```diff
@@ -259,3 +267,52 @@ extension ByteBuffer: AwsStream {
         return !self.status.is_end_of_stream
     }
 }
+
+extension ByteBuffer {
+    public convenience init(data: Data) {
+        self.init(size: data.count)
+        for byte in data {
```

Comment:
Is there not an API to byte swap the whole buffer? There are HUGE accelerations for that if you can do it across whole streams vs byte by byte.

Reply:
see new code, maybe it's faster?